MCLauncher/VersionList.cs

```diff
@@ -49,7 +49,10 @@ public class VersionList : ObservableCollection<WPFDataTypes.Version> {
             // ([name, uuid, versionType])[]
             foreach (JArray o in data.AsEnumerable().Reverse()) {
                 bool isNew = dbVersions.Add(o[0].Value<string>()) && !isCache;
-                Add(new WPFDataTypes.Version(o[1].Value<string>(), o[0].Value<string>(), (WPFDataTypes.VersionType) o[2].Value<int>(), isNew, _commands));
+                int versionType = o[2].Value<int>();
+                if (!Enum.IsDefined(typeof(WPFDataTypes.VersionType), versionType) || versionType == (int) WPFDataTypes.VersionType.Imported)
```

Comment:
isn't `Imported` already a member of `VersionType`?

Reply:
What do you mean by this? I added Imported to VersionType, and this line forbids the server from using the Imported VersionType.

Reply:
My bad, I misunderstood the code.